### ISSUE TITLE
Lost connection to Mono X 6k

### ISSUE THREAD

**Issue by withanhdammit** [adamoutler/anycubic-homeassistant#5]
Not sure why I lost connection to my Mono X 6k.  I uninstalled it then reinstalled it, still no help.

My printer can be seen on the network:
/usr/local/bin/monox.py -i 10.42.100.48 -c sysinfo
sysinfo
model: Photon Mono X 6K
firmware: V0.2.3
serial: 0000141300xxxxxx
wifi: 

Logs show:
`2023-08-20 15:29:49.461 INFO (MainThread) [custom_components.anycubic_wifi.adapter_fascade] Setting up connection
2023-08-20 15:29:49.527 INFO (MainThread) [custom_components.anycubic_wifi.adapter_fascade] Setting up connection
2023-08-20 15:29:49.592 INFO (MainThread) [custom_components.anycubic_wifi.adapter_fascade] Setting up connection
2023-08-20 15:29:49.592 INFO (MainThread) [custom_components.anycubic_wifi.data_bridge] Registering 10.42.100.48
2023-08-20 15:29:49.660 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.anycubic_wifi
2023-08-20 15:29:49.662 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.664 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.664 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.665 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.665 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.666 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.666 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.667 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.668 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.668 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.669 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.669 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.670 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.670 ERROR (MainThread) [homeassistant.components.sensor] Error adding entities for domain sensor with platform anycubic_wifi
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
2023-08-20 15:29:49.671 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up anycubic_wifi platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 370, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 510, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    device = dev_reg.async_get(self.hass).async_get_or_create(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DeviceRegistry.async_get_or_create() got an unexpected keyword argument 'supported_features'
`

**Comment by adamoutler** [comment]
The uart-wifi connection is shoddy on these units.  I don't know why it does it either but sometimes it just stops responding.  I think it's not a very robust protocol.

**Comment by adamoutler** [comment]
Let me rephrase that.  It's obvious that some guy who never worked with protocols designed uart-wifi in an afternoon and made it work enough for the demo where it was accepted.  It's a horrible protocol.

**Comment by bartt-cz** [comment]
Same here, a can paste my logs too, but it's almost same. I think it started with HA 2023.8, so it may be more some HA thing than uart-wifi, but i´m not programmer ...

And anyway, thank you for your work! Until version 2023.8, the library worked flawlessly for many printings

**Comment by adamoutler** [comment]
Fixed in latest release.  V14

**Comment by withanhdammit** [comment]
Thanks @adamoutler!
